test/Sentry.NLog.Tests/SentryTargetTests.cs

```diff
@@ -0,0 +1,340 @@
+                Scope = new Scope(new SentryOptions());
+                Hub.ConfigureScope(Arg.Invoke(Scope));
+            }
+
+            public SentryTarget GetTarget(Action<SentryNLogOptions> customConfig = null)
+            {
+                var options = Options;
+                if (customConfig != null)
+                {
+                    options = new SentryNLogOptions();
+                    customConfig(options);
+                }
+
+                var target = new SentryTarget(
+                    options,
+                    HubAccessor,
+                    SdkDisposeHandle,
+                    Clock)
+                {
+                    Dsn = ValidDsnWithoutSecret,
+                };
+                return target;
+            }
+
+            public LogFactory GetLoggerFactory(Action<SentryNLogOptions> customConfig = null)
+            {
+                var target = GetTarget(customConfig);
+                var config = new LoggingConfiguration();
+                config.AddTarget("sentry", target);
+                config.AddRule(LogLevel.Trace, LogLevel.Fatal, target);
+
+                var factory = new LogFactory(config);
+
+                return factory;
+            }
+
+            public (LogFactory factory, SentryTarget target) GetLoggerFactoryAndTarget(Action<SentryNLogOptions> customConfig = null)
+            {
+                var factory = GetLoggerFactory(customConfig);
+
+                return (factory, factory.Configuration?.FindTargetByName<SentryTarget>("sentry"));
+            }
+
+            public Logger GetLogger(Action<SentryNLogOptions> customConfig = null) => GetLoggerFactory(customConfig).GetLogger("sentry");
+        }
+
+        private readonly Fixture _fixture = new Fixture();
+
+        [Fact]
+        public void Can_configure_from_xml_file()
+        {
+            try
+            {
+                string configXml = $@"
+                <nlog throwConfigExceptions='true'>
+                    <extensions>
+                        <add type='{typeof(SentryTarget).AssemblyQualifiedName}' />
+                    </extensions>
+                    <targets>
+                        <target type='Sentry' name='sentry' dsn='{ValidDsnWithoutSecret}'>
+                            <options>
+                                <environment>Development</environment>
+                            </options>
+                        </target>
+                    </targets>
+                </nlog>";
+
+                var c = XmlLoggingConfiguration.CreateFromXmlString(configXml);
+
+                var t = c.FindTargetByName("sentry") as SentryTarget;
+                Assert.NotNull(t);
+                Assert.Equal(ValidDsnWithoutSecret, t.Options.Dsn.ToString());
+                Assert.Equal("Development", t.Options.Environment);
+            }
+            catch (Exception ex)
+            {
+                _ = ex;
+
+                throw;
```

Comment:
Why catching and rethrowing here?

Reply:
Whoops, that was just a leftover from debugging the test. I'll clean that up.